ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java

```diff
@@ -0,0 +1,134 @@
+   */
+  MySqlReportAggregatorTask(HelixManager manager, long relevantTimePeriodInMs, StatsReportType statsReportType,
+      AccountStatsStore accountStatsStore, Callback<StatsSnapshot> callback, ClusterMapConfig clusterMapConfig) {
+    this(manager, relevantTimePeriodInMs, statsReportType, accountStatsStore, callback, clusterMapConfig,
+        SystemTime.getInstance());
+  }
+
+  MySqlReportAggregatorTask(HelixManager manager, long relevantTimePeriodInMs, StatsReportType statsReportType,
+      AccountStatsStore accountStatsStore, Callback<StatsSnapshot> callback, ClusterMapConfig clusterMapConfig,
+      Time time) {
+    this.manager = manager;
+    clusterAggregator = new HelixClusterAggregator(relevantTimePeriodInMs);
+    this.statsReportType = statsReportType;
+    this.accountStatsStore = accountStatsStore;
+    this.callback = callback;
+    this.clusterMapConfig = clusterMapConfig;
+    this.time = time;
+  }
+
+  @Override
+  public TaskResult run() {
+    Pair<StatsSnapshot, StatsSnapshot> results = null;
+    Exception exception = null;
+    try {
+      List<String> instanceNames = manager.getClusterManagmentTool().getInstancesInCluster(manager.getClusterName());
+      Map<String, StatsWrapper> statsWrappers = new HashMap<>();
+      for (String instanceName : instanceNames) {
+        // Helix instance name would suffix port number, here let's take port number out.
+        instanceName = stripPortNumber(instanceName);
+        statsWrappers.put(instanceName,
+            accountStatsStore.queryStatsOf(clusterMapConfig.clusterMapClusterName, instanceName));
+      }
+      results = clusterAggregator.doWorkOnStatsWrapperMap(statsWrappers, statsReportType);
+      accountStatsStore.storeAggregatedStats(results.getSecond());
+      // Create a base report at the beginning of each month.
+      // Check if there is a base report for this month or not.
+      if (clusterMapConfig.clustermapEnableAggregatedMonthlyAccountReport
+          && statsReportType == StatsReportType.ACCOUNT_REPORT) {
+        // Get the month, if not the same month, then copy the aggregated stats and update the month
+        String currentMonthValue =
+            LocalDateTime.ofEpochSecond(time.seconds(), 0, zoneOffset).format(TIMESTAMP_FORMATTER);
+        String recordedMonthValue = accountStatsStore.queryRecordedMonth(clusterMapConfig.clusterMapClusterName);
+        if (recordedMonthValue == null || recordedMonthValue.isEmpty() || !currentMonthValue.equals(
+            recordedMonthValue)) {
+          accountStatsStore.takeSnapshotOfAggregatedStatsSetMonth(clusterMapConfig.clusterMapClusterName,
+              currentMonthValue);
+        }
+      }
+      return new TaskResult(TaskResult.Status.COMPLETED, "Aggregation success");
+    } catch (Exception e) {
+      logger.error("Exception thrown while aggregating stats from health reports across all nodes ", e);
```

Comment:
let's update this error message to reflect aggregation on mysql based stats report.